File: app/src/main/java/com/scientists/happy/botanist/ui/SimilarPlantsActivity.java
// Show users similar plants
// @author: Christopher Besser and Antonio Muscarella
package com.scientists.happy.botanist.ui;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ListAdapter;
import android.widget.ListView;
import com.scientists.happy.botanist.R;
import com.scientists.happy.botanist.data.DatabaseManager;
public class SimilarPlantsActivity extends AppCompatActivity {
    private DatabaseManager mDatabase;
    /**
     * The activity is launched
     * @param savedInstanceState - current app state
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_similar_plants);
        mDatabase = DatabaseManager.getInstance();
        ListView list = (ListView) findViewById(R.id.similar_plants);
        list.setEmptyView(findViewById(R.id.empty_list_view));
        String group = (String) getIntent().getExtras().get("group");
        String species = (String) getIntent().getExtras().get("species");
        ListAdapter adapter = mDatabase.getSimilarPlants(this, group, species);
        list.setAdapter(adapter);
    }

    /**
     * User navigated up from the activity
     * @return returns true
     */
    @Override
    public boolean onSupportNavigateUp() {
        super.onBackPressed();
        return true;
    }
}

File: app/src/main/java/com/scientists/happy/botanist/ui/SimilarPlantsActivity.kt
// Show users similar plants
// @author: Antonio Muscarella, Christopher Besser and Iskander Gaba
package com.scientists.happy.botanist.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.scientists.happy.botanist.R
import com.scientists.happy.botanist.controller.ActivityController
import com.scientists.happy.botanist.controller.SimilarPlantsController

class SimilarPlantsActivity : AppCompatActivity() {

    private var mController: ActivityController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_similar_plants)
        mController = SimilarPlantsController(this)
    }

    override fun onStart() {
        mController!!.load()
        super.onStart()
    }

    override fun onSupportNavigateUp(): Boolean {
        super.onBackPressed()
        return true
    }
}